### PR TITLE
ESDT parsers support

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Added functions that can determine if a transaction resembles an ESDT / NFT transfer and parse all the wanted data from transaction's payload.

### CODE REVIEW

**Review by raduchis — APPROVED** [review]

File: src/test/java/elrond/ProxyProviderTest.java [hunk 1]
```diff
@@ -26,10 +26,26 @@ public void getNetworkConfig() throws Exception {
...
         assertTrue(account.getNonce() > 0);
         assertTrue(account.getBalance().compareTo(BigInteger.ZERO) > 0);
     }
+
+    @Test
+    public void getESDTBalance() throws Exception {
+        Address address = Address.fromBech32("erd1l453hd0gt5gzdp7czpuall8ggt2dcv5zwmfdf3sd3lguxseux2fsmsgldz");
+        ProxyProvider.ESDTDataResponse esdtData = this.provider.getESDTData(address, "ALC-64e960");
+
+        assertEquals(esdtData.balance, new BigInteger("0"));
```
Comment (iulianpascalau) [inline src/test/java/elrond/ProxyProviderTest.java:41]: can we have a non zero test? Valid also for `getNFTBalance`

Comment (bogdan-rosianu) [reply src/test/java/elrond/ProxyProviderTest.java:41]: These tests are based on a real proxy request. We cannot guarantee that each time a test is run the address will have balance for that ESDT/NFT.

**Review by iulianpascalau — DISMISSED** [review]

File: src/main/java/elrond/ProxyProvider.java [hunk 2]
```diff
@@ -45,13 +47,49 @@ public AccountOnNetwork getAccount(Address address) throws IOException, AddressE
...
+            return ESDTDataResponse.empty();
+        }
+
+        return typedResponse;
+    }
+
+    public BigInteger getNFTBalance(Address address, String tokenIdentifier, long nonce) throws IOException, AddressException {
+        String hexTokenIdentifier = this.adjustTokenIdentifier(tokenIdentifier);
+        String nonceStr = String.format("%d", nonce);
+        if(nonceStr.length() % 2 == 1 ) {
```
Comment (raduchis) [inline src/main/java/elrond/ProxyProvider.java:64]: this check was done with hex, is it needed also for integer strings or should the request be made with hex value?

**Review by raduchis — DISMISSED** [review]

File: src/main/java/elrond/ProxyProvider.java (hunk 2 above)
Comment (bogdan-rosianu) [reply src/main/java/elrond/ProxyProvider.java:64]: Good catch. Fixed

Comment (raduchis) [reply src/main/java/elrond/ProxyProvider.java:64]: ok

**Review by iulianpascalau — DISMISSED** [review]

**Review by raduchis — DISMISSED** [review]

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**